GTG/gtk/preferences.py

```diff
@@ -150,59 +137,56 @@ def _refresh_task_browser(self):
         task_tree.refresh_all()
 
     def activate(self):
-        """ Activate the preferences dialog."""
+        """ Activate the preferences window."""
         self._refresh_preferences_store()
-        self.dialog.show_all()
+        self.window.show_all()
 
-    def valid_check(self, widget):
+    def on_valid_check(self, widget):
         try:
-            self.timer.set_configuration(self.refresh_hour.get_text(),
-                                         self.refresh_mins.get_text())
+            input_time = self.refresh_time.get_text()
+            result_time = self.timer.time_parser(input_time)
+            self.timer.set_configuration(result_time[0], result_time[1])
```

Comment:
Also, when setting time to configuration, you should replace text entry with the parse value. Eg I enter `5:30pm` and when I focus out (click on another button, another window, etc), the text entry should have "17:30" as that is the time saved in configuration